tests/backup/backup_rancher_test.go

```diff
@@ -1,6 +1,192 @@
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Creation and Validation of backup [%s]", backupName))
+			}
+		})
+
+		Step("Restoring to same project but different namespace in same cluster", func() {
+			var restoredNamespaceList []string
+			log.InfoD("Restoring to same project but different namespace in same cluster")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			for _, namespace := range appNamespaces {
+				restoredNamespace := "restored-same-project-" + RandomString(5)
+				namespaceMapping[namespace] = restoredNamespace
+				restoredNamespaceList = append(restoredNamespaceList, restoredNamespace)
+				restoreNamespacesAll = append(restoreNamespacesAll, restoredNamespace)
+				nsObject := &corev1.Namespace{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: restoredNamespace,
+					},
+				}
+				_, err = core.Instance().CreateNamespace(nsObject)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Creating namespace %s", restoredNamespace))
+			}
+			err = Inst().S.(*rke.Rancher).AddNamespacesToProject(sourceClusterProjectList[0], restoredNamespaceList)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Adding namespaces %s to project %s", restoredNamespaceList, sourceClusterProjectList[0]))
+			err = Inst().S.(*rke.Rancher).ValidateProjectOfNamespaces(sourceClusterProjectList[0], restoredNamespaceList)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying project %s of namespace %s", sourceClusterProjectList[0], restoredNamespaceList))
+			restoreName := fmt.Sprintf("%s-same-project-%v", restoreNamePrefix, RandomString(10))
+			restoreList = append(restoreList, restoreName)
+			err = CreateRestore(restoreName, backupName, namespaceMapping, SourceClusterName, orgID, ctx, nil)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Creating restote- %s from backup %s", restoreName, backupName))
+		})
+
+		Step("Restoring to different project and different namespace in same cluster", func() {
+			var restoredNamespaceList []string
+			log.InfoD("Restoring to different project and different namespace in same cluster")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			for _, namespace := range appNamespaces {
+				restoredNamespace := "restored-different-project-" + RandomString(5)
+				namespaceMapping[namespace] = restoredNamespace
+				restoredNamespaceList = append(restoredNamespaceList, restoredNamespace)
+				restoreNamespacesAll = append(restoreNamespacesAll, restoredNamespace)
+				nsObject := &corev1.Namespace{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: restoredNamespace,
+					},
+				}
+				_, err = core.Instance().CreateNamespace(nsObject)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Creating namespace %s", restoredNamespace))
+			}
+			err = Inst().S.(*rke.Rancher).AddNamespacesToProject(sourceClusterProjectList[1], restoredNamespaceList)
```

Comment:
Is it possible to include ValidateProjectOfNamespaces in AddNamespacesToProject itself?

Reply:
It is done on purpose as we might have a requirement in testcase to just check the project of a namespace